storage_devices/lvm.py

```diff
@@ -0,0 +1,544 @@
+
+        cmd = f"vgremove {vg_name}"
+        return TestRun.executor.run(cmd)
+
+    @staticmethod
+    def get_logical_volumes_path(vg_name: str):
+
+        cmd = f"lvdisplay | grep /dev/{vg_name}/ | awk '{{print $3}}'"
+        paths = TestRun.executor.run(cmd).stdout.splitlines()
+
+        return paths if paths else []
+
+
+def get_block_devices_list():
+    cmd = f"lsblk -l | awk '{{print $1}}'"
+    devices = TestRun.executor.run_expect_success(cmd).stdout
+    devices_list = devices.splitlines()
+    devices_list.sort()
+
+    return devices_list
+
+
+class Lvm(Disk):
+    __unique_lv_id = 0
+    __lock = threading.Lock()
+
+    def __init__(
+            self,
+            path: str,
+            volume_group: VolumeGroup,
+            volume_name: str = None
+    ):
+        Device.__init__(self, path)
+        self.device_name = path.split('/')[-1]
+        self.volume_group = volume_group
+        self.volume_name = volume_name
+
+    def __eq__(self, other):
+        try:
+            return self.device_name == other.device_name and \
+                self.volume_group == other.volume_group and \
+                self.volume_name == other.volume_name
+        except AttributeError:
+            return False
+
+    @classmethod
+    def __get_unique_lv_name(cls, prefix: str = "lv"):
+        with cls.__lock:
+            cls.__unique_lv_id += 1
+            return f"{prefix}{cls.__unique_lv_id}"
+
+    @classmethod
+    def __create(
+            cls,
+            name: str,
+            volume_size_cmd: str,
+            volume_group: VolumeGroup
+    ):
+        TestRun.LOGGER.info(f"Creating LV '{name}'.")
+        cmd = f"lvcreate {volume_size_cmd} --name {name} {volume_group.name} --yes"
+        TestRun.executor.run_expect_success(cmd)
+
+        volumes = cls.discover_logical_volumes()
+        for volume in volumes:
+            if name == volume.volume_name:
+                return volume
+
+    @classmethod
+    def configure_global_filter(
+            cls,
+            dev_first: Device,
+            lv_amount: int,
+            pv_devs: ([Device], Device)
+    ):
+        device_first = dev_first.parent_device if isinstance(dev_first, Partition) else dev_first
+        if lv_amount > 1 and isinstance(device_first, Core):
+
+            global_filter_def = LvmConfiguration.read_global_filter_definition_from_lvm_config()
+            if not isinstance(pv_devs, list):
+                pv_devs = [pv_devs]
+
+            if global_filter_def:
+                TestRun.LOGGER.info(f"Configure 'global filter' variable")
+                links = []
+                for pv_dev in pv_devs:
+                    link = pv_dev.get_device_link("/dev/disk/by-id")
+                    links.append(str(link))
+
+                for link in links:
+                    if link in global_filter_def:
+                        TestRun.LOGGER.info(f"Global filter definition already contains '{link}'")
+                        return
```

Comment:
check every link in links
```suggestion
                        continue
```